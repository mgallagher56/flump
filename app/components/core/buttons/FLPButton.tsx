import React from 'react';

import type { ButtonProps } from '@chakra-ui/react';
import { Button } from '@chakra-ui/react';
import * as pressable from '@zag-js/pressable';
import type { PressEvent } from '@zag-js/pressable/dist/pressable.types';
import { normalizeProps, useMachine } from '@zag-js/react';

interface FLPButtonProps extends ButtonProps {
  preventFocusOnPress?: boolean;
  onPress?: (e: PressEvent) => void;
  onLongPress?: (e: PressEvent) => void;
}

const FLPButton: React.FC<React.PropsWithChildren<FLPButtonProps>> = ({
  children,
  colorScheme = 'blue',
  isDisabled,
  preventFocusOnPress,
  variant = 'solid',
  onLongPress,
  onPress,
  isLoading
}) => {
  const [state, send] = useMachine(
    pressable.machine({
      id: 'pressableBaseButton',
      onPress,
      onLongPress,
      disabled: isDisabled,
      preventFocusOnPress
    })
  );

  const api = pressable.connect(state, send, normalizeProps);

  return (
    <Button {...api.pressableProps} variant={variant} colorScheme={colorScheme} isLoading={isLoading}>
      {children}
    </Button>
  );
};

export default FLPButton;
