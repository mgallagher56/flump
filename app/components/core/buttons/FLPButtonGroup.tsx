import React from 'react';

import type { ButtonGroupProps } from '@chakra-ui/react';
import { ButtonGroup } from '@chakra-ui/react';

const FLPButtonGroup: React.FC<React.PropsWithChildren<ButtonGroupProps>> = (props) => {
  return <ButtonGroup {...props}>{props.children}</ButtonGroup>;
};

export default FLPButtonGroup;
