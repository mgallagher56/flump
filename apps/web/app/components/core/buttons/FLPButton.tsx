import { type FC, type PropsWithChildren } from 'react';

import type { ButtonProps } from '@chakra-ui/react';
import { Button } from '@chakra-ui/react';

interface FLPButtonProps extends ButtonProps {
  padding?: number;
}

const FLPButton: FC<PropsWithChildren<FLPButtonProps>> = ({
  colorScheme = 'blue',
  disabled,
  isLoading,
  padding,
  variant = 'solid',
  onClick,
  ...buttonProps
}) => {
  return (
    <Button
      colorScheme={colorScheme}
      isDisabled={disabled}
      isLoading={isLoading}
      padding={padding}
      variant={variant}
      onClick={onClick}
      width={'max-content'}
      {...buttonProps}
    >
      {buttonProps.children}
    </Button>
  );
};

export default FLPButton;
