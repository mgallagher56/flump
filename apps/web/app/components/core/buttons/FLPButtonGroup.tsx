import type { FC, PropsWithChildren } from 'react';

import { Group, type GroupProps } from '@chakra-ui/react';

const FLPButtonGroup: FC<PropsWithChildren<GroupProps>> = (props) => {
  return <Group {...props}>{props.children}</Group>;
};

export default FLPButtonGroup;
