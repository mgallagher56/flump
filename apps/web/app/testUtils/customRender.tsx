// create a custom render function with ChakraProvider
import { type ReactNode } from 'react';

import { type RenderOptions, type RenderResult, render as rtlRender } from '@testing-library/react';
import userEvent, { type UserEvent } from '@testing-library/user-event';
import { Provider } from '~/components/ui/provider';

const user = userEvent.setup();

const customRender = (ui: ReactNode, options?: RenderOptions): RenderResult & { user: UserEvent } => {
  const wrappedUi = <Provider>{ui}</Provider>;

  return {
    ...rtlRender(wrappedUi, options),
    user
  };
};

export default customRender;
