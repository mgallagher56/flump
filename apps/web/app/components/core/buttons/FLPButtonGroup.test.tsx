import { test } from 'vitest';
import customRender from '~/testUtils/customRender';

import FLPButton from './FLPButton';
import FLPButtonGroup from './FLPButtonGroup';

describe('FLPButtonGroup', () => {
  test('renders as expected', () => {
    const { container, getByText } = customRender(
      <FLPButtonGroup>
        <FLPButton>Button 1</FLPButton>
        <FLPButton>Button 2</FLPButton>
      </FLPButtonGroup>
    );
    expect(container).toMatchSnapshot();
    expect(getByText('Button 1')).toBeDefined();
    expect(getByText('Button 2')).toBeDefined();
  });
});
