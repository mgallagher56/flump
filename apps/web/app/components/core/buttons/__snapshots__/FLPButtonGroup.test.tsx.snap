// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FLPButtonGroup > renders as expected 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,r,s,u,d,m,l,h)=&gt;
    {let c=document.documentElement,v=["light","dark"];function p(i){(Array.isArray(e)?e:[e]).forEach(y=&gt;
    {let k=y==="class",S=k&&m?d.map(f=&gt;
    m[f]||f):d;k?(c.classList.remove(...S),c.classList.add(i)):c.setAttribute(y,i)}),R(i)}function R(i){h&&v.includes(i)&&(c.style.colorScheme=i)}function a(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let i=localStorage.getItem(r)||s,y=l&&i==="system"?a():i;p(y)}catch(i){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-6l0f26"
  >
    <button
      class="chakra-button css-1fwa16t"
      data-first=""
      data-group-item=""
      style="--group-count: 2; --group-index: 0;"
      type="button"
    >
      Button 1
    </button>
    <button
      class="chakra-button css-1fwa16t"
      data-group-item=""
      data-last=""
      style="--group-count: 2; --group-index: 1;"
      type="button"
    >
      Button 2
    </button>
  </div>
</div>
`;
